src/core/BaseUtilities.ts

```diff
@@ -271,6 +271,58 @@ export class BaseUtilities {
         return message.replace(regex, match => replacements[match]);
     }
 
+    public async loadDiscordTagData(content: string, guildId: string, cache: DiscordTagSet): Promise<void> {
+        for (const match of content.matchAll(/<[^<>\s]+>/g)) {
+            let id: string | undefined;
+            if ((id = parse.entityId(match[0], '@&')) !== undefined)
+                cache.parsedRoles[id] ??= await this.#getDiscordRoleTag(guildId, id);
+            else if ((id = parse.entityId(match[0], '@!')) !== undefined)
+                cache.parsedUsers[id] ??= await this.#getDiscordUserTag(id);
+            else if ((id = parse.entityId(match[0], '@')) !== undefined)
+                cache.parsedUsers[id] ??= await this.#getDiscordUserTag(id);
```

Comment:
not a fan of a boolean condition with an initialisation in it... Also, couldn't it be merged, regarding the user tag?